guides/source/security.md

```diff
@@ -1361,7 +1361,7 @@ Next, add an initializer to configure the middleware:
 
 ```ruby
 # config/initializers/cors.rb
-Rails.application.config.middleware.insert_before 0, "Rack::Cors" do
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
```

Comment:
This is not outdated

https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/config/initializers/cors.rb.tt#L8

Can you share where you see this?

Reply:
Hey Vipul!

I replaced the `config/initializers/cors.rb` template code with the one from the [documentation](https://guides.rubyonrails.org/security.html#cross-origin-resource-sharing), but I could not start the server.

The [code](https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/config/initializers/cors.rb.tt#L8) that you sent me is the correct version, but it is not the same as the one on the [security guide](https://guides.rubyonrails.org/security.html#cross-origin-resource-sharing). 

my `config/initializers/cors.rb` file looks like this:

```ruby
Rails.application.config.middleware.insert_before 0, "Rack::Cors" do
  allow do
    origins "example.com"

    resource "*",
      headers: :any,
      methods: [:get, :post, :put, :patch, :delete, :options, :head]
  end
end
```

Which is exactly the same from the [docs](https://guides.rubyonrails.org/security.html#cross-origin-resource-sharing), but when I try to running the server it throws the following error:

```sh
/Users/username/.gem/gems/actionpack-7.1.3/lib/action_dispatch/middleware/stack.rb:42:in `build': undefined method `new' for an instance of String (NoMethodError)

        klass.new(app, *args, &block)
             ^^^^
Did you mean?  next
	from /Users/username/.gem/gems/actionpack-7.1.3/lib/action_dispatch/middleware/stack.rb:171:in `block in build'
from /Users/username/.gem/gems/actionpack-7.1.3/lib/action_dispatch/middleware/stack.rb:167:in `each'
	from /Users/username/.gem/gems/actionpack-7.1.3/lib/action_dispatch/middleware/stack.rb:167:in `inject'
	from /Users/username/.gem/gems/actionpack-7.1.3/lib/action_dispatch/middleware/stack.rb:167:in `build'
from /Users/username/.gem/gems/railties-7.1.3/lib/rails/engine.rb:522:in `block in app'
	from /Users/username/.gem/gems/railties-7.1.3/lib/rails/engine.rb:518:in `synchronize'
	from /Users/username/.gem/gems/railties-7.1.3/lib/rails/engine.rb:518:in `app'
from /Users/username/.gem/gems/railties-7.1.3/lib/rails/application/finisher.rb:58:in `block in <module:Finisher>'
	from /Users/username/.gem/gems/railties-7.1.3/lib/rails/initializable.rb:32:in `instance_exec'
	from /Users/username/.gem/gems/railties-7.1.3/lib/rails/initializable.rb:32:in `run'
```


Correct me if I am wrong, also, I believe that using a string on `Rack::Cors` is only possible on Rails versions older than 5.


The correct version to properly initialize the cors is the same as the [template](https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/config/initializers/cors.rb.tt#L8) replacing `"Rack::Cors"` with `Rack::Cors`:

```ruby
Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins "example.com"

    resource "*",
      headers: :any,
      methods: [:get, :post, :put, :patch, :delete, :options, :head]
  end
end
```